x-pack/plugins/observability_solution/entity_manager/server/lib/entities/ingest_pipeline/generate_history_processors.ts

```diff
@@ -144,6 +170,24 @@ export function generateHistoryProcessors(definition: EntityDefinition) {
         ignore_missing: true,
       },
     },
+    { script: { source: createIdentityFieldsPainlessScript(definition) } },
```

Comment:
is it necessary to have this complex logic vs something more simple like 

```
    ...definition.identityFields.map((key) => {
      return {
        set: {
          field: key.field,
          value: `{{entity.identityFields.${key.field}}}`,
          if: `ctx.entity?.identityFields?.${key.field.replaceAll('.', '?.')} != null`,
        }
      }
    }),
```

does this correctly handle the nested object behaviour?

Reply:
did this get tested? i think it should work?

Reply:
Yes, it works! Put it back into the original PR!